lib/ecto/postgres/pg_rational_migrations.ex

```diff
@@ -0,0 +1,132 @@
+use Vtc.Ecto.Postgres.Utils
+
+defpgmodule Vtc.Ecto.Postgres.PgRational.Migrations do
+  @moduledoc """
+  Migrations for adding rational types, functions and constraints to a
+  Postgres database.
+  """
+  alias Ecto.Migration
+
+  require Ecto.Migration
+
+  @doc section: :migrations_full
+  @doc """
+  Adds raw SQL queries to a migration for creating the database types, associated
+  functions, casts, operators, and operator families.
+
+  Includes all migraitons under the
+  [PgTypes](Vtc.Ecto.Postgres.PgRational.Migrations.html#pgtypes) and
+  [PgFunctions](Vtc.Ecto.Postgres.PgRational.Migrations.html#pgfunctions)
+  headings.
+
+  Safe to run multiple times when new functionality is added in updates to this library.
+  Existing values will be skipped.
+
+  ## Types Created
+
+  Calling this macro creates the following type definitions:
+
+  ```sql
+  CREATE TYPE public.rational AS (
+    numerator bigint,
+    denominator bigint
+  );
+  ```
+
+  ## Schemas Creates
+
+  Two schemas are created to house our native rational functions:
+
+  - `rationals`: Holds "public", user-facing functions.
+  - `rationals_helpers`: Holds "private" helper functions our "public" functions in the
+    `rationals` schema rely on.
+
+  ## Functions Creates
+
+  See `ecto_functions` section of these docs for details on native database functions
+  created.
+
+  ## Examples
+
+  ```elixir
+  defmodule MyMigration do
+    use Ecto.Migration
+
+    alias Vtc.Ecto.Postgres.PgRational
+    require PgRational.Migrations
+
+    def change do
+      PgRational.Migrations.create_all()
+    end
+  end
+  ```
+  """
+  @spec create_all() :: :ok
+  def create_all do
+    :ok = create_type()
+
+    :ok
+  end
+
+  @doc section: :migrations_types
+  @doc """
+  Adds:
+
+  - `rational` composite type
+  - `rationals` schema
+  - `rationals_helpers` schema
```

Comment:
```suggestion
  - `rational` composite type
```

Reply:
In 7433851362ebc3aa1c0e8109020424ba83ad6d0a